include/djinterop/engine/v2/engine_library.hpp

```diff
@@ -0,0 +1,133 @@
+/*
+    This file is part of libdjinterop.
+
+    libdjinterop is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    libdjinterop is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with libdjinterop.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#if __cplusplus < 201703L
+#error This library needs at least a C++17 compliant compiler
+#endif
+
+#include <memory>
+#include <string>
+
+#include <djinterop/config.hpp>
+#include <djinterop/database.hpp>
+#include <djinterop/engine/engine_version.hpp>
+#include <djinterop/engine/v2/change_log_table.hpp>
+#include <djinterop/engine/v2/information_table.hpp>
+#include <djinterop/engine/v2/playlist_entity_table.hpp>
+#include <djinterop/engine/v2/playlist_table.hpp>
+#include <djinterop/engine/v2/track_table.hpp>
+
+namespace djinterop::engine::v2
+{
+struct engine_library_context;
```

Comment:
The `engine_library_context` struct is forward-declared and hence unavailable to clients.  It acts somewhat like a pimpl for the low-level API in that it:

1. Prevents the user from constructing `_table` classes directly and instead requires them to use `engine_library` as the gateway.
2. Hides the implementation details of _how_ the library interacts with the library on disk.  Although it is public knowledge that there is a SQLite database in a given directory, the decision by the library of _how_ to interact with that SQLite database is considered an implementation detail.